### ISSUE TITLE
gauge 1.1.1 / gauge-java 0.7.8 : launcher.sh contains bash-ism but points to #!/bin/sh

### ISSUE THREAD

**Issue by kimholan** [getgauge/gauge#1687]
### Describe the bug

~/.gauge/plugins/java/0.7.8/launcher.sh contains bashisms which breaks the launcher plugin when /bin/sh
is a symlink to a non-bash shell such as /bin/dash

### What command(s) did you run when you found the bug?

running gauge specs inside a container based on debian:buster-slim which defaults to dash for /bin/sh

Example code from launcher.sh as found in '~/.gauge/plugins/java/0.7.8/launcher.sh '

```
remove_substr_from_string() {
    string=$1
    substr=$2
    echo ${string/$substr/}
}

```

### Output, stack trace or logs related to the bug 

Source the remove_substr_from_string inside bash works as expected, calling the shell functions gives 'd1do' as output:

```bash
$ /bin/bash
$ remove_substr_from_string() {
>     string=$1
>     substr=$2
>     echo ${string/$substr/}
> }

$ remove_substr_from_string d1ldo l
d1do
```

When the function is sourced inside /bin/dash, it fails:

```dash
$ /bin/dash
$ remove_substr_from_string() {
    string=$1
    substr=$2
    echo ${string/$substr/}
}
> > > > $ remove_substr_from_string d1ldo l
/bin/dash: 4: Bad substitution
```

### Versions 

0.7.8

#### Gauge (Output of `gauge -v`)

```
$ gauge -v
Gauge version: 1.1.1
Commit Hash: 6e0d83f

Plugins
-------
html-report (4.0.11)
java (0.7.8)
screenshot (0.0.1)
xml-report (0.2.3)
```

#### Node.js/Java/Python/.Net/Ruby version

```
$ java -version
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
```

#### Operating System information

```
$ uname -a
Linux debian 5.4.0-0.bpo.3-amd64 #1 SMP Debian 5.4.13-1~bpo10+1 (2020-02-07) x86_64 GNU/Linux
```

#### IDE information

none. run from shell.

**Comment by kimholan** [comment]
See https://wiki.ubuntu.com/DashAsBinSh for an short summary of differences that can break when shell scripts use bash specific functionality when /bin/sh is not bash.

**Comment by sriv** [comment]
thanks for investigating and reporting the issue with shebang.

We introduced the shell/powershell script launcher recently, but it's causing some issues. The plan is to revert to the old launcher (with some changes), which uses golang.

Some work has been pushed to getgauge/gauge-java#455/

**Comment by BugDiver** [comment]
This should be fixed with release 0.7.9. This release remove use of shell scripts and create golang binaries.
